### PR TITLE
docs: allow empty targets tensor in ctc_loss

### LINKED ISSUE #150995: `torch.nn.functional.ctc_loss` inconsistent implementation and docs
### 📚 The doc issue

When fixing #150835 find out here's an inconsistency between doc and test of `torch.nn.functional.ctc_loss`.

The [doc](https://pytorch.org/docs/stable/generated/torch.nn.functional.ctc_loss.html) describe `targets` param `cannot be blank`

Image

But there's a test case in `test_nn.py` do validate pass empty tensor as `targets` and compute loss result.

https://github.com/pytorch/pytorch/blob/4273e5d15cfcb282b2795684874ea439d8620999/test/test_nn.py#L11413-L11421

So it's confused whether `targets` is allowed to be blank or not.

### Suggest a potential alternative/fix

May change the doc description about `targets` param

cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki

### PR DESCRIPTION
**Pull request by jPorterDosch**
docs: allow empty targets tensor in ctc_losswhen target_lengths are zero, as described in issue 

Fixes #150995

### CODE REVIEW

**Review by albanD — APPROVED** [review]
Thanks!

**Comment by jPorterDosch** [comment]
@albanD fixed the line length issue that was causing the automated tests to fail.

**Comment by cyyever** [comment]
@jPorterDosch Fix linter

**Comment by jPorterDosch** [comment]
@cyyever Done.
